site-scrapers/PriceChopper.js

```diff
@@ -0,0 +1,75 @@
+const sites = require("../data/sites.json");
+const https = require("https");
+const crypto = require('crypto');
+
+const siteName = 'Price Chopper';
+const site = sites[siteName];
+
+module.exports = async function GetAvailableAppointments(browser) {
+    console.log(`${siteName} starting.`);
+    const webData = await ScrapeWebsiteData(browser);
+    console.log(`${siteName} done.`);
+    return site.locations.map((loc) => {
+        const locHash = md5HashString(loc.street + loc.city);
+        const responseLocation = webData[locHash];
+        let hasAvailability = false;
+        let availability = {};
+        if(responseLocation && responseLocation.visibleTimeSlots) {
+            hasAvailability = responseLocation.visibleTimeSlots.length ? true : false;
+            for(let [i, timeSlot] of responseLocation.visibleTimeSlots.entries()) {
+                let date = timeSlot.time.split('T')[0];
+                if(!availability[date]){
+                    availability[date] = {
+                        hasAvailability: true,
+                        numberAvailableAppointments: 0,
+                        signUpLink: site.signUpLink
+                    };
+                }
+                availability[date].numberAvailableAppointments += 1;
+            }
+        }
+        return {
+            name: `${siteName} (${loc.city})`,
+            hasAvailability,
+            availability,
+            signUpLink: site.signUpLink, // or https://pdhi.queue-it.net/?c=pdhi&e=covid19vaccination&t_portalUuid=3e419790-81a3-4639-aa08-6bd223f995df
```

Comment:
So I checked today and this link goes to the same place. Could you please replace it? It'll save our users an extra step (a hard one - the link to sign up for an appointment is pretty hidden.

Reply:
No prob. All for making things easier. We should probably discuss some automated testing so that we know the links we use still work across all the sites.